### PR TITLE
ORC-452: Support converting JSON object to MAP type

### PR DESCRIPTION
**Pull request by stiga-huang**
Add `MapColumnConverter` to support converting JSON object to MAP type.
Since JSON keys can only be strings, we only support MAP type with string keys.

### CODE REVIEW

**Comment by stiga-huang** [comment]
@omalley Would you please have a look for this.

**Comment by stiga-huang** [comment]
Hi @wgtmac, do you have time to review this? Is there anyone I can invite too?

File: java/tools/src/java/org/apache/orc/tools/convert/JsonReader.java [hunk 1]
```diff
@@ -222,6 +225,40 @@ public void convert(JsonElement value, ColumnVector vect, int row) {
     }
   }
 
+  static class MapColumnConverter implements JsonConverter {
+    private JsonConverter keyConverter;
+    private JsonConverter valueConverter;
+
+    public MapColumnConverter(TypeDescription schema) {
+      TypeDescription keyType = schema.getChildren().get(0);
+      if (keyType.getCategory() != TypeDescription.Category.STRING)
```
Comment (wgtmac) [inline java/tools/src/java/org/apache/orc/tools/convert/JsonReader.java]: Why it can't support other types?

Comment (wgtmac) [reply java/tools/src/java/org/apache/orc/tools/convert/JsonReader.java]: Got it. Thanks!

**Pull request merged**